src/main/java/frc/robot/hardware/mechanisms/SimulatedAccelerometer.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.hardware.mechanisms;
+
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.hardware.interfaces.IAccelerometer;
+import frc.utils.alerts.Alert;
+import frc.utils.alerts.AlertManager;
+import frc.utils.alerts.PeriodicAlert;
+import frc.utils.time.TimeUtils;
+
+import java.util.Random;
+import java.util.function.Supplier;
+
+public class SimulatedAccelerometer implements IAccelerometer {
+
+	private final Supplier<ChassisSpeeds> robotSimulatedVelocityRelativeToRobot;
+	private final Supplier<Double> noise;
+	private final String logPath;
+
+	private double accelerationX;
+	private double accelerationY;
+	private ChassisSpeeds previousRobotVelocities;
+	private double lastUpdateTimestamp;
+
+	public SimulatedAccelerometer(String logPath, Supplier<ChassisSpeeds> robotSimulatedVelocity, double noiseAmount) {
+		Random randomNumbersGenerator = new Random();
+		this.logPath = logPath;
+		this.robotSimulatedVelocityRelativeToRobot = robotSimulatedVelocity;
+		this.noise = () -> randomNumbersGenerator.nextGaussian() * noiseAmount;
```

Comment:
You do not need a refrence of the Random objcet
```suggestion
		this.logPath = logPath;
		this.robotSimulatedVelocityRelativeToRobot = robotSimulatedVelocity;
		this.noise = () -> new Random.nextGaussian() * noiseAmount;
```

Reply:
i don't want to recalc the seed each time. performance